python/idsse_common/idsse/common/protocol_utils.py

```diff
@@ -0,0 +1,173 @@
+
+        Returns:
+            Sequence[datetime]: A sequence of issue date/times
+        """
+        zero_time_delta = timedelta(seconds=0)
+        if time_delta == zero_time_delta:
+            raise ValueError('Time delta must be non zero')
+
+        issues_set: set[datetime] = set()
+        if issue_start:
+            datetimes = datetime_gen(issue_end, time_delta, issue_start, num_issues)
+        else:
+            # check if time delta is positive, if so make negative
+            if time_delta > zero_time_delta:
+                time_delta = timedelta(seconds=-1.0 * time_delta.total_seconds())
+            datetimes = datetime_gen(issue_end, time_delta)
+        for issue_dt in datetimes:
+            if issue_start and issue_dt < issue_start:
+                break
+            try:
+                dir_path = self.path_builder.build_dir(issue=issue_dt)
+                issues = {self.path_builder.get_issue(file_path)
+                          for file_path in self.ls(dir_path)
+                          if file_path.endswith(self.path_builder.file_ext)}
+                issues_set.update(issues)
+                if num_issues and len(issues_set) >= num_issues:
+                    break
+            except PermissionError:
+                pass
+        if None in issues_set:
+            issues_set.remove(None)
+        return sorted(issues_set)[:num_issues]
+
+    def get_valids(self,
+                   issue: datetime,
+                   valid_start: datetime | None = None,
+                   valid_end: datetime | None = None) -> Sequence[tuple[datetime, str]]:
+        """Get all objects consistent with the passed issue date/time and filter by valid range
+
+        Args:
+            issue (datetime): The issue date/time used to format the path to the object's location
+            valid_start (datetime | None, optional): All returned objects will be for
+                                              valids >= valid_start. Defaults to None.
+            valid_end (datetime | None, optional): All returned objects will be for
+                                            valids <= valid_end. Defaults to None.
+
+        Returns:
+            Sequence[tuple[datetime, str]]: A sequence of tuples with valid date/time (indicated by
+                                            object's location) and the object's location (path).
+                                            Empty Sequence if no valids found for given time range.
+        """
+        if valid_start and valid_start == valid_end:
+            valids_and_filenames = self.check_for(issue, valid_start)
+            return [valids_and_filenames] if valids_and_filenames is not None else []
+
+        dir_path = self.path_builder.build_dir(issue=issue)
+        valid_and_file = [(self.path_builder.get_valid(file_path), file_path)
+                          for file_path in self.ls(dir_path)
+                          if file_path.endswith(self.path_builder.file_ext)]
```

Comment:
I think there is a bug here, that http_utils exposes. It's might be possible to get file_paths that don't match with the specified issueDt. When I tried adding the following suggestion it cause problems with wildcard, so maybe something different and/or changes elsewhere.
```suggestion
                          if file_path.endswith(self.path_builder.file_ext) and
                          self.path_builder.get_issue(file_path] == issue]
```